scipy/optimize/optimize.py

```diff
@@ -776,12 +777,13 @@ def _line_search_wolfe12(f, fprime, xk, pk, gfk, old_fval, old_old_fval,
 
     if ret[0] is None:
         # line search failed: try different one.
-        with warnings.catch_warnings():
-            warnings.simplefilter('ignore', LineSearchWarning)
-            kwargs2 = {}
-            for key in ('c1', 'c2', 'amax'):
-                if key in kwargs:
-                    kwargs2[key] = kwargs[key]
+        kwargs2 = {}
+        for key in ('c1', 'c2', 'amax'):
+            if key in kwargs:
+                kwargs2[key] = kwargs[key]
+        with suppress_warnings() as sup:
+            sup.filter(LineSearchWarning,
+                       "The line search algorithm did not converge")
```

Comment:
Yes, I'm not sure about this change --- `suppress_warnings` does all sorts trawling through modules resetting warning registries, which may not be so great inside the innards of an optimization algorithm.

I would leave this one as it was, and silence it in the tests where it appears. (The actual issue here is probably that the LineSearchWarning is emitted in the wrong place, sounds like it should be emitted by the *user* of the linesearch, not by the linesearch itself.)